nixos/modules/services/mail/public-inbox.nix

```diff
@@ -0,0 +1,569 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.public-inbox;
+  stateDir = "/var/lib/public-inbox";
+
+  manref = name: vol: "<citerefentry><refentrytitle>${name}</refentrytitle><manvolnum>${toString vol}</manvolnum></citerefentry>";
+
+  singleIniAtom = with types; nullOr (oneOf [ bool int float str ]) // {
+    description = "INI atom (null, bool, int, float or string)";
+  };
+  iniAtom = with types; coercedTo singleIniAtom singleton (listOf singleIniAtom) // {
+    description = singleIniAtom.description + " or a list of them for duplicate keys";
+  };
+  iniAttrs = with types; attrsOf (either (attrsOf iniAtom) iniAtom);
+  gitIni = {
+    type = with types; attrsOf iniAttrs;
+    generate = name: value: pkgs.writeText name (generators.toGitINI value);
+  };
+
+  environment = {
+    PI_EMERGENCY = "${stateDir}/emergency";
+    PI_CONFIG = gitIni.generate "public-inbox.ini"
+      (filterAttrsRecursive (n: v: v != null) cfg.settings);
+  };
+
+  useSpamAssassin = cfg.settings.publicinboxmda.spamcheck == "spamc" ||
+                    cfg.settings.publicinboxwatch.spamcheck == "spamc";
+
+  serviceConfig = srv: {
+    # Enable JIT-compiled C (via Inline::C)
+    Environment = [ "PERL_INLINE_DIRECTORY=/run/public-inbox-${srv}/perl-inline" ];
+    # NonBlocking is REQUIRED to avoid a race condition
+    # if running simultaneous services.
+    NonBlocking = true;
+    #LimitNOFILE = 30000;
+    User = config.users.users."public-inbox".name;
+    Group = config.users.groups."public-inbox".name;
+    RuntimeDirectory = [
+      "public-inbox-${srv}/perl-inline"
+      # Create RootDirectory= in the host's mount namespace.
+      "public-inbox-${srv}/root"
+    ];
+    RuntimeDirectoryMode = "700";
+    # Avoid mounting RootDirectory= in the own RootDirectory= of ExecStart='s mount namespace.
+    InaccessiblePaths = ["-+/run/public-inbox-${srv}/root"];
+    # This is for BindPaths= and BindReadOnlyPaths=
+    # to allow traversal of directories they create in RootDirectory=.
+    UMask = "0066";
+    RootDirectory = "/run/public-inbox-${srv}/root";
+    RootDirectoryStartOnly = true;
+    WorkingDirectory = stateDir;
+    MountAPIVFS = true;
+    BindReadOnlyPaths = [
+      builtins.storeDir
+      "/etc"
+      "/run"
+    ];
+    BindPaths = [
+      stateDir
+    ];
+    # The following options are only for optimizing:
+    # systemd-analyze security public-inbox-'*'
+    AmbientCapabilities = "";
+    CapabilityBoundingSet = "";
+    # ProtectClock= adds DeviceAllow=char-rtc r
+    DeviceAllow = "";
+    LockPersonality = true;
+    MemoryDenyWriteExecute = true;
+    NoNewPrivileges = true;
+    PrivateDevices = true;
+    PrivateMounts = true;
+    PrivateNetwork = mkDefault false;
+    PrivateTmp = true;
+    PrivateUsers = true;
+    ProtectClock = true;
+    ProtectControlGroups = true;
+    ProtectHome = true;
```

Comment:
This should be at least a `mkDefault`. I have coderepos in /home that public-inbox needs to be able to see.

Reply:
Done. I've also added latest available options: `ProtectProc = "noaccess" ` and `ProcSubset = "pid";`